### PR TITLE
Create user form

### PR DESCRIPTION
**Pull request by arbulu89**
# Description

Add the `Create User` view to `/users/new`.
Nothing fancy:
- Some basic validation in the frontend to confirm that required fields are present
- Show validation errors if they happen

To do:
- The password generation button. We need to have a look on this
- There is not any error handling for optimistic locking, or anything that is not related to the validation by now (anyway, this shouldn't affect to the user creation in any case)

Have a look on storybook to see the available scenarios.

image

## How was this tested?

Tested with UT and manually

### CODE REVIEW

**Review by abravosuse — APPROVED** [review]
Userform component reviewed in storybook. It looks good to me.

**Review by jagabomb — APPROVED** [review]
LGTM!

**Comment by arbulu89** [comment]
Just opening for review, i will defer the merge until the `List` component is merged first

**Review by rtorrero — APPROVED** [review]

File: assets/js/pages/Users/UserForm.jsx [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+          </div>
+          <p className="col-span-6">
+            <span className="text-red-500">*</span> Required Fields
+          </p>
+        </div>
+        <div className="flex flex-row w-80 space-x-2 mt-5">
+          <Button
+            disabled={saving}
+            type="default-fit"
+            onClick={() => {
```
Comment (CDimonaco) [inline assets/js/pages/Users/UserForm.jsx:207]: Could we move that to a dedicated function?

File: assets/js/pages/Users/UserForm.jsx [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+                return;
+              }
+
+              const user = {
+                fullname: fullNameState,
+                email: emailAddressState,
+                username: usernameState,
+                password: passwordState,
+                password_confirmation: confirmPasswordState,
+                enabled: statusState === 'Enabled',
```
Comment (CDimonaco) [inline assets/js/pages/Users/UserForm.jsx:218]: Could we have 'Enabled' as a const?

**Review by CDimonaco — COMMENTED** [review]
LGTM,

I left some comments, overall very good job!

**Review by EMaksy — APPROVED** [review]
Hey nice job, lgtm, left some minor questions/suggestions!

File: assets/js/pages/Users/CreateUserPage.test.jsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+
+import { render, screen } from '@testing-library/react';
+import 'intersection-observer';
+import '@testing-library/jest-dom';
+
+import userEvent from '@testing-library/user-event';
+import MockAdapter from 'axios-mock-adapter';
+
+// eslint-disable-next-line import/no-extraneous-dependencies
```
[the remark this answers is not in the document]
Comment (arbulu89) [reply assets/js/pages/Users/CreateUserPage.test.jsx:10]: We could do it yes.
Feel free to create a tech-debt ticket :stuck_out_tongue:

File: assets/js/pages/Users/CreateUserPage.test.jsx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+const usersUrl = '/api/v1/users';
+const axiosMock = new MockAdapter(networkClient);
+
+describe('CreateUserPage', () => {
+  beforeEach(() => {
+    axiosMock.reset();
+    jest.spyOn(console, 'error').mockImplementation(() => null);
+  });
+
+  it('Back To Users redirects to the users view', async () => {
```
[the remark this answers is not in the document]
Comment (arbulu89) [reply assets/js/pages/Users/CreateUserPage.test.jsx:28]: Ok, i will change the descriptions

File: assets/js/pages/Users/CreateUserPage.test.jsx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+// eslint-disable-next-line import/no-extraneous-dependencies
+import * as router from 'react-router';
+
+import { networkClient } from '@lib/network';
+
+import { userFactory } from '@lib/test-utils/factories/users';
+
+import CreateUserPage from './CreateUserPage';
+
+const usersUrl = '/api/v1/users';
```
[the remark this answers is not in the document]
Comment (arbulu89) [reply assets/js/pages/Users/CreateUserPage.test.jsx:19]: Ok, I will change it

File: assets/js/pages/Users/UserForm.jsx [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+  password = '',
+  confirmPassword = '',
+  status = 'Enabled',
+  errors = defaultErrors,
+  saving = false,
+  saveText = 'Create',
+  onSave = noop,
+  onCancel = noop,
+}) {
+  const [fullNameState, setFullName] = useState(fullName);
```
[the remark this answers is not in the document]
Comment (EMaksy) [reply assets/js/pages/Users/UserForm.jsx:45]: Yeah, I see why you added the suffix :+1:  it's fine for me!

**Pull request merged**